server/rest/src/main/java/org/infinispan/rest/RestServer.java

```diff
@@ -186,4 +195,28 @@ public List<CorsConfig> getCorsConfigs() {
       }
       return corsRules;
    }
+
+   public CompletionStage<List<NodeDistributionInfo>> clusterDistribution() {
+      List<Address> members = cacheManager.getMembers();
```

Comment:
We can't use members like this. It seems you are doing that to calculate when the consumers are all completed. However, this has a small window when a new member could join or leave that could cause it to miss an update or even never complete. 

Instead you should chain off the CompletionStage returned from the `submitConsumer` method to handle this.

Reply:
I've updated another place where I was doing the same thing. Thanks!!